components/tsi-util/secret-maker.sh

```diff
@@ -0,0 +1,207 @@
+#!/bin/bash
+
+SECRET_VALUE='$SECRET_VALUE'
+
+### Functions:
+
+# Print the header:
+header()
+{
+cat << EOF
+#!/bin/bash
+
+# **** Trusted Service Identity ****
+# **** IBM Research ****************
+# https://github.com/IBM/trusted-service-identity/
+#
+# This auto-generated script provides helpful tooling
+# for injecting TSI secrets into Vault based on the input
+# deployment file
+#
+# Assumptions:
+#  1. File has properly defined annotations including
+#    a. 'tsi.secrets'
+#    b. 'admission.trusted.identity/inject: "true"'
+#  2.
+
+
+# Define values for the SECRET_VALUE(s)
+# they must be in "key=value" format
+# For more info see:
+# https://github.com/IBM/trusted-service-identity/examples/vault/README.md#secrets
+export SECRET_VALUE="secret=xxx"
+
+# To access vault, obtain the Vault Client from (...)
+#  and define ROOT_TOKEN and VAULT_ADDR env. variables:
+#
+# export ROOT_TOKEN=
+# export VAULT_ADDR=(vault address in format http://vault.server:8200)
+#
+EOF
+
+# prevent expression expending by using a single quote:
+echo 'vault login -no-print "${ROOT_TOKEN}"
+RT=$?
+if [ $RT -ne 0 ] ; then
+   echo "ROOT_TOKEN is not set correctly"
+   exit 1
+fi'
+
+}
+
+## create help menu:
+helpme()
+{
+  cat <<HELPMEHELPME
+This script helps building TSI policies
+
+syntax:
+   $0 [cluster-info.yaml] [pod-info.yaml]
+where:
+      [cluster-info.yaml] - cluster info, otherwise defaults to '/tmp/clusterinfo'
+      [pod-info.yaml] - pod info, otherwise defaults to '/tmp/podinfo'
+HELPMEHELPME
+}
+
+# buildSecrets - build individual secret entries
+buildSecrets()
+{
+  local SECNAME=$1
+  local CONSTR=$2
+
+  case $CONSTR in
+      "region")
+          echo "# using policy $CONSTR"
+          echo "vault kv put secret/tsi-r/${REGION}/${SECNAME} ${SECRET_VALUE}"
+          POLICIES+=('tsi-r')
+          ;;
+      "region,images")
+          echo "# using policy $CONSTR"
+          echo "vault kv put secret/tsi-ri/${REGION}/${IMGSHA}/${SECNAME} ${SECRET_VALUE}"
+          POLICIES+=('tsi-ri')
+          ;;
+      "region,cluster-name,namespace")
+          echo "# using policy $CONSTR"
+          ROLE="tsi-role-rcn"
+          VAULT_PATH="secret/tsi-rcn"
+          echo "vault kv put secret/tsi-rcn/${REGION}/${CLUSTER}/${NS}/${SECNAME} ${SECRET_VALUE}"
+          POLICIES+=('tsi-rcn')
+          ;;
+      "region,cluster-name,namespace,images")
+          echo "# using policy $CONSTR"
+          echo "vault kv put secret/tsi-rcni/${REGION}/${CLUSTER}/${NS}/${IMGSHA}/${SECNAME} ${SECRET_VALUE}"
+          POLICIES+=('tsi-rcni')
+          ;;
+      *) echo "# ERROR: invalid constraints requested: ${CONSTR}"
+         ;;
```

Comment:
return an error code here? So that it can propagate error and exit?

Reply:
This error would not qualitfy for terminating the script. Error is displayed in comments of the script and the secret would not be processed.